lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php

```diff
@@ -303,6 +304,16 @@ protected function validateRuntimeMetadata($class, $parent)
         }
     }
 
+    private function validateAssociationTargets(ClassMetadata $class): void
+    {
+        foreach ($class->getAssociationMappings() as $mapping) {
+            $targetEntity = $mapping['targetEntity'];
+            if ($this->driver->isTransient($targetEntity) || $this->peekIfIsMappedSuperclass($targetEntity)) {
```

Comment:
Is this correct, or am I missing a case where different drivers might be used for entities from different modules or similar? 

There is just one ClassMetadataFactory for all entities (within the boundaries of a particular entity manager), and it has a single driver, right?